File: fn/src/test/java/org/sqlbuilder/fn/TestNormalizer.java
package org.sqlbuilder.fn;

import org.junit.Test;

public class TestNormalizer
{
	@Test
	public void normalize1()
	{
		System.out.println("\n-- normalize ferealizations (feid)");
		new Normalizer("Fn_fes", "fetypeid", "feid") //
				.targets("Fn_ferealizations", "fetypeid", "feid") //
				.referenceThrough("Fn_lexunits", "luid", "Fn_frames", "frameid") //
				.dump();
	}

	@Test
	public void normalize2()
	{
		System.out.println("\n-- normalize fegrouprealizations (feid)");
		new Normalizer("Fn_fes", "fetypeid", "feid") //
				.targets("Fn_fegrouprealizations_fes", "fetypeid", "feid") //
				.referenceThrough("Fn_fegrouprealizations", "fegrid", "Fn_lexunits", "luid", "Fn_frames", "frameid") //
				.dump();
	}

	@Test
	public void normalize3()
	{
		System.out.println("\n-- normalize lexunits (incorporatedfeid)");
		new Normalizer("Fn_fes", "fetypeid", "feid") //
				.targets("Fn_lexunits", "incorporatedfetypeid", "incorporatedfeid") //
				.referenceThrough("Fn_frames", "frameid") //
				.dump();
	}

	@Test
	public void normalize4()
	{
		System.out.println("\n-- normalize grouppatterns_patterns (feid)");
		new Normalizer("Fn_fes", "fetypeid", "feid") //
				.targets("Fn_grouppatterns_patterns", "fetypeid", "feid") //
				.referenceThrough("Fn_patterns", "patternid", "Fn_fegrouprealizations", "fegrid", "Fn_lexunits", "luid") //
				.dump();
	}
}

File: fn/src/test/java/org/semantikos/fn/TestNormalizer.kt
package org.semantikos.fn

import org.junit.Test

class TestNormalizer {

    @Test
    fun normalize1() {
        println("\n-- normalize ferealizations (feid)")
        Normalizer("Fn_fes", "fetypeid", "feid")
            .targets("Fn_ferealizations", "fetypeid", "feid")
            .referenceThrough("Fn_lexunits", "luid", "Fn_frames", "frameid")
            .dump()
    }

    @Test
    fun normalize2() {
        println("\n-- normalize fegrouprealizations (feid)")
        Normalizer("Fn_fes", "fetypeid", "feid")
            .targets("Fn_fegrouprealizations_fes", "fetypeid", "feid")
            .referenceThrough("Fn_fegrouprealizations", "fegrid", "Fn_lexunits", "luid", "Fn_frames", "frameid")
            .dump()
    }

    @Test
    fun normalize3() {
        println("\n-- normalize lexunits (incorporatedfeid)")
        Normalizer("Fn_fes", "fetypeid", "feid")
            .targets("Fn_lexunits", "incorporatedfetypeid", "incorporatedfeid")
            .referenceThrough("Fn_frames", "frameid")
            .dump()
    }

    @Test
    fun normalize4() {
        println("\n-- normalize grouppatterns_patterns (feid)")
        Normalizer("Fn_fes", "fetypeid", "feid")
            .targets("Fn_grouppatterns_patterns", "fetypeid", "feid")
            .referenceThrough("Fn_patterns", "patternid", "Fn_fegrouprealizations", "fegrid", "Fn_lexunits", "luid")
            .dump()
    }
}
